app/v2/services/RetrieveHistoricNonFhlUkPropertyAnnualSubmissionService.scala

```diff
@@ -21,42 +21,43 @@ import uk.gov.hmrc.http.HeaderCarrier
 import utils.Logging
 import v2.connectors.RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector
 import v2.controllers.EndpointLogContext
-import v2.models.errors.{InternalError, MtdError, NinoFormatError, NotFoundError, TaxYearFormatError}
+import v2.models.errors.{ InternalError, MtdError, NinoFormatError, NotFoundError, RuleHistoricTaxYearNotSupportedError, TaxYearFormatError }
 import v2.models.request.retrieveHistoricNonFhlUkPropertyAnnualSubmission.RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequest
 import v2.models.response.retrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse.RetrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse
 import v2.support.DownstreamResponseMappingSupport
 
-import javax.inject.{Inject, Singleton}
-import scala.concurrent.{ExecutionContext, Future}
+import javax.inject.{ Inject, Singleton }
+import scala.concurrent.{ ExecutionContext, Future }
 
 @Singleton
 class RetrieveHistoricNonFhlUkPropertyAnnualSubmissionService @Inject()(connector: RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector)
-  extends DownstreamResponseMappingSupport
-  with Logging {
-
-    private val downstreamErrorMap: Map[String, MtdError] =
-      Map(
-        "INVALID_NINO"        -> NinoFormatError,
-        "INVALID_TYPE"        -> InternalError,
-        "INVALID_TAX_YEAR"    -> TaxYearFormatError,
-        "NOT_FOUND_PROPERTY"  -> NotFoundError,
-        "NOT_FOUND_PERIOD"    -> NotFoundError,
-        "SERVER_ERROR"        -> InternalError,
-        "SERVICE_UNAVAILABLE" -> InternalError
-      )
-
-    def retrieve(request: RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequest)(
+    extends DownstreamResponseMappingSupport
+    with Logging {
+
+  private val downstreamErrorMap: Map[String, MtdError] =
```

Comment:
Can we move the **downstreamErrorMap**  into a companion object and then import into both service and tests to avoid duplication of errorMappings in both code + tests : https://github.com/hmrc/property-business-api/blob/b3d5974e1602261d3e282933b6fe473fc2e2a82a/app/v2/services/RetrieveHistoricNonFhlUkPropertyPeriodSummaryService.scala#L54-L67